comp/forwarder/defaultforwarder/domain_forwarder_test.go

```diff
@@ -79,9 +79,15 @@ func TestDomainForwarderStop(t *testing.T) {
 	forwarder.Stop(false) // this should be a noop
 	forwarder.Start()
 	assert.Equal(t, Started, forwarder.State())
+
+	// Add a transaction which should be flushed to retry queue on stop
+	tr := transaction.NewHTTPTransaction()
+	forwarder.requeuedTransaction <- tr
+	requireLenForwarderRetryQueue(t, forwarder, 0)
```

Comment:
This test appears to be flaky:
```
    domain_forwarder_test.go:356:
        	Error Trace:	/go/src/github.com/DataDog/datadog-agent/comp/forwarder/defaultforwarder/domain_forwarder_test.go:356
        	            				/go/src/github.com/DataDog/datadog-agent/comp/forwarder/defaultforwarder/domain_forwarder_test.go:86
        	Error:      	Not equal:
        	            	expected: 0
        	            	actual  : 1
        	Test:       	TestDomainForwarderStop
```

Reply:
It was tricky to find a way to assert that we're flushing transactions that hadn't yet made it into the retry queue, so if this way is flaky then I'll probably just remove it. I'm open to any suggestions on that, but it may just not be that vital of a thing to test given that it's not the primary goal anyway.